lib/variable/to_unit.cpp

```diff
@@ -29,8 +29,12 @@ Variable to_unit(const Variable &var, const units::Unit &unit,
                  const CopyPolicy copy) {
   if (unit == var.unit())
     return copy == CopyPolicy::Always ? variable::copy(var) : var;
-  const auto scale = llnl::units::quick_convert(
-      variableFactory().elem_unit(var).underlying(), unit.underlying());
+  const auto var_unit = variableFactory().elem_unit(var);
+  if (var_unit == units::none)
+    throw except::UnitError(
+        "Conversion from a `none` unit to another unit is not permitted.");
```

Comment:
- Refer to `None`, not `none`.
- I would not call it "none unit", `None` means there is no unit.

Reply:
"Unit conversion is not permitted as the input variable has no unit." ?

Reply:
Something like that.